crates/polars-core/src/frame/column/scalar.rs

```diff
@@ -307,3 +307,71 @@ impl From<ScalarColumn> for Column {
         Self::Scalar(value)
     }
 }
+
+#[cfg(feature = "serde")]
+mod serde_impl {
+    use std::sync::OnceLock;
+
+    use polars_error::PolarsError;
+    use polars_utils::pl_str::PlSmallStr;
+
+    use super::ScalarColumn;
+    use crate::frame::{Scalar, Series};
+
+    #[derive(serde::Serialize, serde::Deserialize)]
+    struct SerializeWrap {
+        name: PlSmallStr,
+        /// Unit-length series for dispatching to IPC serialize
+        unit_series: Series,
```

Comment:
For `ScalarColumn` I had the option of either using the `serde::Serialize` impl from `AnyValue`, or converting it to a unit-length Series and dispatch to IPC. I picked the IPC option as the `AnyValue` serde impl was missing quite a lot of dtypes, and using IPC would also give more assurance that the serialize behavior is the same with the `SeriesColumn`.